### ISSUE TITLE
Add a rule to validate message's examples

### ISSUE THREAD

**Issue by magicmatatjahu** [stoplightio/spectral#2125]
**User story.**
As a user I want to know if the examples defined in the message object are valid.

**Is your feature request related to a problem?**
The following AsyncAPI document is invalid and Spectral should throw errors:

```yaml
asyncapi: "2.0.0"
info:
  title: Bad API
  version: "1.0.0"
components:
  messages:
    someMessage:
      payload:
        type: string
      headers:
        type: object
      examples:
        - payload: 2137 # should be string
          headers: someHeader # should have object shape
```

**Additional context**
Part of #2100
